tests/system/Database/ForgeTest.php

```diff
@@ -0,0 +1,69 @@
+<?php namespace CodeIgniter\Database;
+
+use CodeIgniter\Database\MockConnection;
+
+class QueryTest extends \CIUnitTestCase
+{
+
+	protected $db;
+
+	//--------------------------------------------------------------------
+
+	public function setUp()
+	{
+		$this->db = new MockConnection([]);
+	}
+
+	//--------------------------------------------------------------------
+
+	public function testAddForeignKey()
+	{
+	    $forge = new Forge($this->db);
+	    
+	    $fk[] = [
+	        'key' => 'key',
+	        'fkey' => 'fkey',
+	        'table' => 'table',
+	        'ftable' => 'ftable',
+	        'onUpdate' => 'CASCADE',
+	        'onDelete' => 'CASCADE'
+	    ];
+	    
+	    $forge->addForeignKey('key', 'fkey', 'table', 'ftable', 'CASCADE', 'CASCADE');
+
+		$this->assertEquals($fk, $forge->getForeingKeys());
+	}
+	
+	//-------------------------------------------------------------------
+	
+	public function testCompileFkeyCreate()
+	{
+	    $forge = new Forge($this->db);
+	    
+	    $fk = [
+	            'key' => 'key',
+	            'fkey' => 'fkey',
+	            'table' => 'table',
+	            'ftable' => 'ftable',
+	            'onUpdate' => 'CASCADE',
+	            'onDelete' => 'CASCADE'
+	    ];
+	    
+	    $forge->addForeignKey('key', 'fkey', 'table', 'ftable', 'CASCADE', 'CASCADE');
+	     
+	    $sql = 'ALTER TABLE table ADD CONSTRAINT fk_fkey FOREIGN KEY (key) REFERENCES ftable (fkey) ON UPDATE CASCADE ON DELETE CASCADE';
```

Comment:
Thank you for your comment, Mr. Whitney. The change has being done. Can't get out from those other files. Thanks.